### DISCUSSION TITLE
Teste admissional prático, pode ser feito em JS?

### DISCUSSION THREAD

**Discussion by rudineyliemes** [frontendbr/forum#1887] [Perguntas e Respostas]
Bom dia a todos,

Estou em uma empresa no ramo de seguros, porem ainda não estou trabalhando na área de TI, estou estudando HTML/CSS/JS e na minha empresa o teste admissional é em Java e C#.
Minha duvida seria o teste abaixo pode ser feito em JS e se sim por onde começo, gostaria muito de realizar em JS na linguagem que estou estudando e não partir para outras.

segue abaixo teste.
----------------------------------------------//--------------------------------------------------------------------------------//----------------------------------------------------------------
Suponha que a administração do prédio 99a da Tecnopuc, com 16 andares e cinco elevadores, denominados A, B, C, D e E, nos convidou a aperfeiçoar o sistema de controle dos elevadores. Depois de realizado um levantamento no qual cada usuário respondia:
a. O elevador que utiliza com mais frequência (A, B, C, D ou E);
b. O andar ao qual se dirigia (0 a 15);
c. O período que utilizava o elevador – M: Matutino; V: Vespertino; N: Noturno.

Considerando que este possa evoluir para um sistema dinâmico, escreva o código que nos ajude a extrair as seguintes informações:
a. Qual é o andar menos utilizado pelos usuários;
b. Qual é o elevador mais frequentado e o período que se encontra maior fluxo;
c. Qual é o elevador menos frequentado e o período que se encontra menor fluxo;
d. Qual o período de maior utilização do conjunto de elevadores;
e. Qual o percentual de uso de cada elevador com relação a todos os serviços prestados;

Deve ser programado em Java ou C#.

Para a realização do exercício você deve implementar a interface IElevadorService.

Faça a leitura do arquivo input.json para ter acesso às entradas.

**Comment by evelew** [comment]
Oi @rudineyliemes, acho que é melhor você tirar essa dúvida com a galera da empresa que você ta fazendo o teste, assim você não corre o risco de fazer em JS e eles invalidarem o teste.

**Comment by dbtrnl** [comment]
Melhor tentar ver com quem ofereceu a vaga mesmo. Java e C# usam vários conceitos que no JS sequer existem ou só nas versões mais recentes (ES6+) tão implementando.

Como curiosidade, tem esse jogo em JS que você implementa o código que controla um elevador e tem que cumprir várias tarefas. Talvez seja útil!

https://play.elevatorsaga.com/

**Comment by rudineyliemes** [comment]
vlw galera , estou vendo c na faculdade, irei ver C# logo mais e assim poder realizar esse teste.
